setup.py

```diff
@@ -11,7 +11,7 @@
 #                Bumping Minor means API bugfix or new functionality.
 #                Bumping Micro means CLI change of any kind unless it is
 #                    significant enough to warrant a minor/major bump.
-version = '3.7.0'
+version = '3.8.0'
```

Comment:
Since this is a pretty big behavior change, lets bump it to 4.0.0, so its easier for people to pin to <4.0.0 if they want the polling and caching behavior.